src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouppedObjects = {};
+
+  items.forEach((item) => {
+    const resultKey = String(item[key]);
+
+    if (grouppedObjects[resultKey]) {
+      grouppedObjects[resultKey].push(item);
+    } else {
+      grouppedObjects[resultKey] = [item];
+    }
+
+    return grouppedObjects;
```

Comment:
it looks like you wanted to use reduce, but used forEach instead, did you? Why did you need the second parameter for forEach?

Reply:
My bad, forgot to remove🤦

Reply:
there is also no need to return something from the callback of foreach